### DISCUSSION TITLE
Publish Panel support a message defined by server and send it to server ?

### DISCUSSION THREAD

**Discussion by chaosLeee** [foxglove/community#804] [Q&A] [answered]
hi , im trying to use the publish panel to  send a message which has been defined in my foxglove websocket server, but i find that panel must connected to a ros source( in  https://foxglove.dev/docs/studio/panels/publish#user-interactions ), how can i send any type message in the studio ? thx for your deep concern

**Comment by achim-k** [comment, marked as answer]
The publish panel docs that you linked to are a bit outdated, sorry for the confusion. You can use the publish panel with any live data source that supports publishing, it doesn't have to be necessarily ROS. When using the foxglove websocket connection, the publish panel can be used when your server advertises the [`clientPublish`](https://github.com/foxglove/ws-protocol/blob/main/docs/spec.md#server-info) capability.
